packages/playwright-core/src/server/browserContext.ts

```diff
@@ -232,9 +232,12 @@ export abstract class BrowserContext extends SdkObject {
       // at the same time.
       return;
     }
+    const gotClosedGracefully = this._closedStatus === 'closing';
     this._closedStatus = 'closed';
-    this._deleteAllDownloads();
-    this._downloads.clear();
+    if (!gotClosedGracefully) {
```

Comment:
This way if the browser gets killed when browser.close() started (e.g. during tracing.dispose()) we may end up not deleting the files at all.

I don't understand how it helps with the test flakiness. AFAIU the problem is that we don't wait for the deletions to complete here but assume it in the test.

Reply:
As discussed offline let's remove these two lines completely. If the browser is killed by a signal we'll leak the artifacts but it should be fine as it may happen in other scenarios where the browser is killed too.